### PR TITLE
Add test runner and test suite for emulating user commands

### PR DESCRIPTION
**Pull request by Brainicism**

### CODE REVIEW

File: src/test/end-to-test/test-runner-bot.ts [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+    }
+
+    console.log(
+        `=====================STAGE ${CURRENT_STAGE}===================`,
+    );
+
+    const stageData = BASIC_OPTIONS_TEST_SUITE.tests[CURRENT_STAGE]!;
+    const command = stageData.command;
+    if (TEST_SUITE.resetEachStage) {
+        await sendCommand(",reset");
```
Comment (taahamahdi) [inline src/test/end-to-test/test-runner-bot.ts:86]: Should this use the prefix

File: src/test/end-to-test/test_suites/basic_options_test.ts [hunk 2]
```diff
@@ -0,0 +1,237 @@
+import type ParsedGameOptionValues from "../parsed_game_options_value";
+import type TestSuite from "./test_suite";
+
+const BASIC_OPTIONS_TEST_SUITE: TestSuite = {
+    tests: [
+        {
+            command: ",limit 2",
```
Comment (taahamahdi) [inline src/test/end-to-test/test_suites/basic_options_test.ts:7]: These should use bot prefix

**Review by taahamahdi — APPROVED** [review]
Some suggestions but looks good

File: src/helpers/discord_utils.ts [hunk 3]
```diff
@@ -439,6 +439,21 @@ export async function sendMessage(
     authorID?: string,
     interaction?: Eris.ComponentInteraction | Eris.CommandInteraction,
 ): Promise<Eris.Message | null> {
+    if (messageContent.messageReference) {
+        const message: Message<TextChannel> | undefined = await (
+            State.client.getChannel(textChannelID!) as
+                | Eris.TextChannel
+                | undefined
+        )?.getMessage(messageContent.messageReference.messageID);
```
Comment (taahamahdi) [inline src/helpers/discord_utils.ts:447]: Will this lookup slow down non-test send message calls

Comment (Brainicism) [reply src/helpers/discord_utils.ts:447]: These are cached

File: src/test/end-to-test/test_suites/basic_options_test.ts (hunk 2 above)
Comment (Brainicism) [reply src/test/end-to-test/test_suites/basic_options_test.ts:7]: They are replaced with bot prefix in the runner code

File: src/test/end-to-test/test-runner-bot.ts (hunk 1 above)
Comment (Brainicism) [reply src/test/end-to-test/test-runner-bot.ts:86]: It's replaced in sendCommand

**Pull request merged**